### ISSUE TITLE
Invariant Violation: ReactCompositeComponent: injectEnvironment() can only be called once.

### ISSUE THREAD

**Issue by shenrenguo** [storybooks/storyshots#45]
sorry, not provide sufficient information.

**.storybook/config.js**

```
import React from 'react';
import { configure, addDecorator } from '@kadira/storybook';
import centered from '@kadira/react-storybook-decorator-centered';
import '../src/common.scss';

addDecorator(centered);

const context = require.context('../stories', true, /_story\.js$/);

function loadStories() {
  context.keys().forEach(context);
}

configure(loadStories, module);
```

**.storybook/webpack.config.js**

```
const path = require('path');
const webpack = require('webpack');
const autoprefixer = require('autoprefixer');
const bourbon = require('node-bourbon');

function includePaths() {
  return bourbon.includePaths;
}

module.exports = {
  module: {
    loaders: [
      {
        test: /\.js$/,
        exclude: /(node_modules)/,
        loader: 'babel'
      },
      {
        test: /\.json$/,
        loader: 'json',
      },
      {
        test: /\.scss$/,
        loader: 'style!css?sourceMap&modules&importLoaders=1&localIdentName=[name]__[local]__[hash:base64:5]!postcss!sass',
        include: path.resolve(__dirname, '../')
      }
    ]
  },
  postcss: [autoprefixer],
  sassLoader: {
    data: '@import "' + path.resolve(__dirname, '../src/_theme.scss') + '";',
    includePaths: includePaths()
  }
};
```

**package.json** :  

```
"dependencies": {
    "bourbon": "^4.2.7",
    "immutability-helper": "^2.0.0",
    "normalize.css": "^4.2.0",
    "react": "^15.3.2",
    "react-addons-css-transition-group": "^15.2.1",
    "react-addons-test-utils": "^15.2.1",
    "react-addons-update": "^15.3.2",
    "react-css-themr": "^1.1.3",
    "react-dom": "^15.3.2",
    "react-intl": "^2.1.3",
    "react-toolbox": "^1.2.0"
  },
  "devDependencies": {
    "@kadira/react-storybook-decorator-centered": "^1.0.0",
    "@kadira/storybook": "^2.0.0",
    "@kadira/storyshots": "^2.0.1",
    "autoprefixer": "^6.3.7",
    "babel-cli": "^6.11.4",
    "babel-core": "^6.11.4",
    "babel-eslint": "^6.1.2",
    "babel-loader": "^6.0.1",
    "babel-preset-es2015": "^6.1.4",
    "babel-preset-react": "^6.1.4",
    "babel-preset-stage-0": "^6.5.0",
    "babel-register": "^6.3.13",
    "bumped": "^0.9.2",
    "chai": "^3.5.0",
    "chai-enzyme": "^0.5.1",
    "cheerio": "^0.22.0",
    "classnames": "^2.2.3",
    "cpx": "^1.3.2",
    "cross-env": "^2.0.0",
    "css-loader": "^0.23.1",
    "enzyme": "^2.4.1",
    "eslint": "^2.13.1",
    "eslint-config-airbnb": "^9.0.1",
    "eslint-plugin-import": "^1.11.0",
    "eslint-plugin-jsx-a11y": "^1.5.5",
    "eslint-plugin-react": "^5.2.2",
    "extract-text-webpack-plugin": "^1.0.1",
    "jsdoc": "^3.4.0",
    "json-loader": "^0.5.4",
    "karma": "^1.3.0",
    "karma-chrome-launcher": "^1.0.1",
    "karma-cli": "^1.0.1",
    "karma-firefox-launcher": "^1.0.0",
    "karma-mocha": "*",
    "karma-mocha-reporter": "^2.1.0",
    "karma-phantomjs-launcher": "^1.0.1",
    "karma-safari-launcher": "^1.0.0",
    "karma-sourcemap-loader": "^0.3.7",
    "karma-spec-reporter": "0.0.26",
    "karma-tap-reporter": "0.0.6",
    "karma-webpack": "^1.7.0",
    "lodash": "^4.14.1",
    "minimatch": "^3.0.2",
    "mocha": "^3.1.0",
    "node-bourbon": "^4.2.8",
    "node-sass": "^3.8.0",
    "node-static": "^0.7.8",
    "normalize.css": "^4.2.0",
    "postcss-loader": "^0.9.1",
    "rimraf": "^2.5.4",
    "sass-loader": "^4.0.0",
    "sinon": "^2.0.0-pre.2",
    "sinon-chai": "^2.8.0",
    "style-loader": "^0.13.1",
    "supports-color": "^3.1.2",
    "webpack": "^1.13.1",
    "webpack-dev-middleware": "^1.6.1",
    "webpack-hot-middleware": "^2.12.1"
  },
  "scripts": {
    "build": "cross-env NODE_ENV=production npm run build:js && npm run build:scss",
    "build:clean": "rimraf ./lib",
    "build:js": "babel ./src --out-dir ./lib",
    "build:scss": "cpx './src/**/*.scss' ./lib -v",
    "lint": "npm run lint:js && npm run lint:scss",
    "lint:js": "eslint ./ --ext .js",
    "lint:scss": "scss-lint '**/*.scss' -v -e 'node_modules/**/*.scss'",
    "patch": "bumped release patch",
    "prebuild": "npm run build:clean",
    "prepublish": "npm run build",
    "release": "bumped release",
    "storybook": "start-storybook -p 9001 -s ./public",
    "storybook:build": "npm run storybook:clean && build-storybook -o storybook-build -s ./public",
    "storybook:clean": "rimraf ./storybook-build",
    "storybook:start": "static -p 8001 storybook-build",
    "test": "cross-env NODE_ENV=test karma start",
    "test:watch": "cross-env NODE_ENV=test karma start --no-single-run",
    "test:storybook": "storyshots"
  }

```

script cmd :->  "storybook": "start-storybook -p 9001 -s ./public"  can work correctly. but
when i run script cmd :-> "test:storybook": "storyshots", it shows error message below:

=> Loading custom .babelrc
Invariant Violation: ReactCompositeComponent: injectEnvironment() can only be called once.
    at invariant (/Users/srg/workspaces/TENMIC/reactui/node_modules/fbjs/lib/invariant.js:38:15)
    at Object.injectEnvironment (/Users/srg/workspaces/TENMIC/reactui/node_modules/react/lib/ReactComponentEnvironment.js:36:60)
    at Object.inject (/Users/srg/workspaces/TENMIC/reactui/node_modules/react/lib/ReactDefaultInjection.js:79:28)
    at Object.<anonymous> (/Users/srg/workspaces/TENMIC/reactui/node_modules/react/lib/ReactDOM.js:28:23)
    at Module._compile (module.js:573:32)
    at Module._extensions..js (module.js:582:10)
    at Object.require.extensions.(anonymous function) [as .js](/Users/srg/workspaces/TENMIC/reactui/node_modules/babel-register/lib/node.js:152:7)
    at Module.load (module.js:490:32)
    at tryModuleLoad (module.js:449:12)
    at Function.Module._load (module.js:441:3)

**Comment by arunoda** [comment]
Not sure what's happening here.
We need a way to reproduce locally.
May be a sample project?

**Comment by arunoda** [comment]
Or the steps to reproduce this.

**Comment by 0xR** [comment]
I have the same issue when I run `npm run test-storybook`.

My repo: https://github.com/xebia/styled-react-app

**Comment by SpaceK33z** [comment]
I can also reproduce this. It happens when you use `require.context` in `.storybook/config.js`. The docs use this snippet:

``` js
import { configure } from '@kadira/storybook';

const req = require.context('../src/components', true, /.stories.js$/)

function loadStories() {
  req.keys().forEach((filename) => req(filename))
}

configure(loadStories, module);
```

And that works, as long as you have only one file that ends with `.stories.js`. As soon as there is more than one file, you'll get the `injectEnvironment` error.

**Comment by SpaceK33z** [comment]
Debugged this a bit more, it isn't an issue in `require.context` after all. It also happens when require-ing two or more story files:

``` js
import { configure } from '@kadira/storybook';

function loadStories() {
    require('../src/components/Button.stories');
    require('../src/components/Bar.stories');
}

configure(loadStories, module);
```

**Comment by arunoda** [comment]
cc @roonyh

**Comment by roonyh** [comment]
It should be related to https://github.com/facebook/react/issues/7386

We [replace](https://github.com/storybooks/storyshots/blob/master/bin/run_storyshots.js#L10) react-dome with a mock react-dom as a workaround to this problem. I guess somehow thats not working here.

@SpaceK33z Do you use react-dom api in your stories?

**Comment by SpaceK33z** [comment]
The issue seems to be fixed with React 15.4.0 :). My multiple stories run again via storyshots.
